47.md

```diff
@@ -63,6 +63,238 @@ Response:
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "description": "some description", // invoice description, optional
+        "description_hash": "31afdf1..", // invoice description hash, optional
+        "preimage": "0123456789abcdef...", // preimage for the payment, optional
+        "expiry": 3600, // invoice expiry in seconds, optional
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "make_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "invoice": "lnbc50n1...",
+        "payment_hash": "31afdf1.."
+    }
+}
+```
+
+### Lookup Invoice
+
+Request:
+```jsonc
+{
+    "method": "lookup_invoice",
+    "params": {
+        "payment_hash": "31afdf1..", // payment hash of the invoice, one of payment_hash or invoice is required
+        "invoice": "lnbc50n1..." // invoice to lookup
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "lookup_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "invoice": "lnbc50n1...", // the encoded invoice
+        "paid": true // true if the invoice has been paid
+    }
+}
+```
+
+
+### List Invoices
+
+Request:
+```jsonc
+{
+    "method": "list_invoices",
+    "params": {
+        "from": 1693876973, // starting timestamp in seconds since epoch, optional
+        "until": 1693876973, // ending timestamp in seconds since epoch, optional
+        "limit": 10, // maximum number of invoices to return, optional
+        "offset": 0, // offset of the first invoice to return, optional
+        "unpaid": true // include unpaid invoices, optional, default false
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "list_invoices",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": [
+        {
+           invoice: "string", // encoded invoice
+           payee_pubkey: "string", // pubkey of the payee, used for keysend
+           memo: "string",
+           description_hash: "string",
+           preimage: "string",
+           payment_hash: "string",
+           amount: 123, // value in msats
+           fees_paid: 123, // value in msats
+           settled_at: unixtimestamp,
+           metadata: {} // generic metadata that can be used to add things like zap/boostagram details for a payer name/comment/etc.
+        }
+    ],
+}
+```
+
+### List Payments
+
+Request:
+```jsonc
+{
+    "method": "list_payments",
+    "params": {
+        "from": 1693876973, // starting timestamp in seconds since epoch, optional
+        "until": 1693876973, // ending timestamp in seconds since epoch, optional
+        "limit": 10, // maximum number of invoices to return, optional
+        "offset": 0, // offset of the first invoice to return, optional
```

Comment:
would it make sense to have a param here to include unsettled payments, similar to `unpaid` in `list_invoices`?

Reply:
what would an unpaid payment be? only thing that'd make sense is failed or in progress but dont think that really fits here

Reply:
Yes not `unpaid` but a similar idea, as you said, failed and in-progress payments (for example HOLD invoices).

Why does it not fit here? would it make sense as a future improvement?

Reply:
lets just do that later, think this has plenty of functionality